src/Compilers/Core/Portable/CommandLine/CommonCompiler.SuppressionDiagnostic.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal abstract partial class CommonCompiler
+    {
+        /// <summary>
+        /// Special informational diagnostic for each programmatic <see cref="Diagnostics.Suppression"/> reported by a <see cref="Diagnostics.DiagnosticSuppressor"/>.
+        /// </summary>
+        private sealed class SuppressionDiagnostic : Diagnostic
+        {
+            private static readonly DiagnosticDescriptor s_suppressionDiagnosticDescriptor = new DiagnosticDescriptor(
+                "SP0001",
+                CodeAnalysisResources.SuppressionDiagnosticDescriptorTitle,
+                CodeAnalysisResources.SuppressionDiagnosticDescriptorMessage,
+                "ProgrammaticSuppression",
+                DiagnosticSeverity.Info,
+                isEnabledByDefault: true);
+
+            private readonly Diagnostic _originalDiagnostic;
+            private readonly string _suppressionId;
+            private readonly LocalizableString _suppressionJustification;
+
+            public SuppressionDiagnostic(
+                Diagnostic originalDiagnostic,
+                string suppressionId,
+                LocalizableString suppressionJustification)
+            {
+                Debug.Assert(originalDiagnostic != null);
+                Debug.Assert(originalDiagnostic.ProgrammaticSuppressionInfo != null);
+                Debug.Assert(!string.IsNullOrEmpty(suppressionId));
+                Debug.Assert(suppressionJustification != null);
+
+                _originalDiagnostic = originalDiagnostic;
+                _suppressionId = suppressionId;
+                _suppressionJustification = suppressionJustification;
+            }
+
+            public override DiagnosticDescriptor Descriptor => s_suppressionDiagnosticDescriptor;
+
+            public override string Id => Descriptor.Id;
+
+            public override string GetMessage(IFormatProvider formatProvider = null)
+            {
+                // Diagnostic '{0}: {1}' was programmatically suppressed by a DiagnosticSuppressor with suppresion ID '{2}' and justification '{3}'
+                return string.Format(formatProvider,
+                    CodeAnalysisResources.SuppressionDiagnosticDescriptorMessage,
```

Comment:
>CodeAnalysisResources.SuppressionDiagnosticDescriptorMessage [](start = 20, length = 60)

This should take ```formatProvider``` (CultureInfo) into account. See how compiler loads resources for the given CultureInfo.

Reply:
I mimic'ed the implementation here: http://source.roslyn.io/#Microsoft.CodeAnalysis/Diagnostic/Diagnostic_SimpleDiagnostic.cs,95